lib/index.js

```diff
@@ -100,6 +100,8 @@ module.exports = function(options) {
         jwt.verify(token, secret, options, function(err, decoded) {
           if (err) {
             callback(new UnauthorizedError('invalid_token', err));
+          } else if(decoded.id != req.headers.id) {
+            callback(new UnauthorizedError('invalid_token', err));
```

Comment:
`err` will be undefined at this point. Presumably you want to indicate `token-id does not match declared header-id`?